### ISSUE TITLE
[java] UnusedPrivateField: false positive with `j++`

### ISSUE THREAD

**Issue by abcHarrison** [pmd/pmd#4381]
**Affects PMD Version:** 6.52.0

**Rule:**
UnusedPrivateField

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedprivatefield>

**Description:**
As the example shown in the rule documentation, `return j++` means the usage of private field `j`. If I change it to `j++`, PMD report `UnusedPrivateField:     Avoid unused private fields such as 'j'`. While if I change it to `j=j+1`, which has the same semantics to `j++`, PMD will not give any report of `j`. Interestingly, `++j` has no violation as well.
**Code Sample demonstrating the issue:**
Maybe a false positive :

```java
public class Something {
    private int j = 6; //false positive here
    public void addOne() {
         j++; 
    }
}
```

Compliant cases:

```java
public class Something {
    private int j = 6;
    public void addOne() {
         j=j+1; 
    }
}
```
```java
public class Something {
    private int j = 6;
    public void addOne() {
        ++j;                   
    }
}
```
```java
public class Something {
    private int j = 6;
    public int addOne() {
        return j++; 
    }
}
```

**Expected outcome:**
No warning.

PMD reports a violation at line 2, but that's wrong. That's a false positive.

**Running PMD through:** *[CLI]*

**Comment by jsotuyod** [comment]
I don't think this is a false positive at all. `j` is being modified, but it's never actually being read for anything at all (it's not returned, passed as a parameter or used in an rvalue expression). In the provided code the field can be completely removed and no visible behavior would be altered.

On the contrary, I'd argue that the case:

```java
public class Something {
    private int j = 6;
    public void addOne() {
        ++j;                   
    }
}
```

is actually a false-negative if not being reported.

The other two scenarios are actually ok, even if the first one (`j = j + 1`) is still bogus, from a codebase standpoint, the value is being read to compute an expression.

Having said that, PMD 7.0.0-rc3 actually behaves according to this definition, with the standalone increment unary operators being flagged as unused.
